lints/lint_rsa_public_exponent_not_in_range.go

```diff
@@ -27,12 +27,16 @@ func (l *rsaParsedTestsExpInRange) CheckApplies(c *x509.Certificate) bool {
 }
 
 func (l *rsaParsedTestsExpInRange) RunTest(c *x509.Certificate) (ResultStruct, error) {
-	pubKey := c.PublicKey.(*rsa.PublicKey).E
+	key, ok := c.PublicKey.(*rsa.PublicKey)
+	if !ok {
+		return ResultStruct{Result: Error}, nil
+	}
+	exponent := key.E	
 	lowerBound := 65536 // 2^16 + 1
 	var upperBound big.Int
 	upperBound.Exp(big.NewInt(2), big.NewInt(256), nil)
 
-	if pubKey > lowerBound && upperBound.Cmp(big.NewInt(int64(pubKey))) == 1 {
+	if exponent > lowerBound && upperBound.Cmp(big.NewInt(int64(exponent))) == 1 {
```

Comment:
I'm super confused by this lint now that I read it. IIUC, this will fail if exponent is < 65536? There are lots of perfectly fine exponents lower than that. Like 3.